chain-core/src/property.rs

```diff
@@ -91,11 +91,17 @@ pub trait Block: Serialize + Deserialize {
 /// block's metadata via a network protocol or in other uses where the
 /// full content of the block is too bulky and not necessary.
 pub trait HasHeader {
+    /// The block header id.
+    type Id: BlockId;
+
     /// The block header type.
     type Header: Header;
 
     /// Retrieves the block's header.
     fn header(&self) -> Self::Header;
+
+    /// Retrieves the block's header id.
+    fn id(&self) -> Self::Id;
```

Comment:
This seems redundant as the same method exists on trait `Block`. Instead, I think the `Header` trait should have this method.

Reply:
but the header is smaller than the block, doesn't it? The idea is that we want to get an `id` having only Header, but not the block itself.
At the same time `Block` trait doesn't imply `Header` trait so header trait method is not enough as well.

So as I see we need both, but maybe I should rename the method for header?

Reply:
why can't the trait `Header` provide the `id` method?

have it duplicated on both trait for now (Block and Header). This is easier.